app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -376,6 +381,17 @@ class _CreatePinConsumerState extends ConsumerState<CreatePinPage> {
     }
   }
 
+  Future<String?> selectSpace() async {
```

Comment:
so considering that this comes in here twice, this should probably be moved into a helper action, should it not?